src/dhtproto/client/UsageExamples.d

```diff
@@ -26,6 +26,15 @@
 
 module dhtproto.client.UsageExamples;
 
+
+/// Mixin helper for the SmartUnion's final switch handlers. It inserts
+/// case none: assert(false); in D2 and default: assert(false); in D1
+istring handleInvalidCases ()
```

Comment:
This should be placed in a convenient place so that user can reach it. Ideally, it should go to `SmartUnion`. I don't think ocean's dependency matters here.

Reply:
Yes, you could make a PR to ocean with this as well.

Reply:
If this code is always compiled with D2 as well then D1 could use `case none`, too -- `default` is only needed as a safety guard if an enum case has been forgotten, but the D2 build will now fail in that case.

Reply:
Do you mean that the purpose of `case none: ` in D1 as well is to prevent duplicate `case none: ` that might be injected by the user?

Reply:
The duplicate would be there in D2 then, but that was not my point.
I understand that `default` is needed in D1 to avoid a warning that a `switch` has no `default`. But then I would always inject `case none:`, add `default:` for D1 merely to suppress the warning, and document that, to keep D1 and D2 code as similar as possible.

Reply:
For example:
```d
"case none: assert(false);" ~
// The D2 build makes sure all enum cases are covered so in fact we don't
// need a default. However, D1 emits a compiler warning if a switch
// has no default so we add a default here just to avoid the warning.
"version (D_Version2){} else {default: assert(false);}"
```

Reply:
Could you add a TODO here to update this to use the code in ocean, when the new version is released.

Reply:
Updated.